### DISCUSSION TITLE
Join as complex relationsip with write_only strategy gives incorrect error

### DISCUSSION THREAD

**Discussion by artofhuman** [sqlalchemy/sqlalchemy#12114] [Usage Questions] [answered]
```python
class Company(Base):
   id: Mapped[int_pk]
   inn: Mapped[str]


class Document(Base):
   id: Mapped[int_pk]
   org_id: Mapped[int] = mapped_column(ForeignKey("company.id"))
   is_active: Mapped[int]


def _wide_active_documents_rel():
    orgs = alias(Company)
    documents = join(Document, orgs, Document.org_id == orgs.c.id)

    return relationship(
        aliased(Document, documents, flat=True),
        viewonly=True,
        uselist=True,
        lazy="write_only",
        primaryjoin=foreign(Company.inn) == orgs.c.inn
    )


Company.documents = _wide_active_documents_rel()
```

If I use `lazy='write_only'` it raises an error 
```
sqlalchemy.exc.InvalidRequestError: On relationship Company.documents, 'dynamic' loaders cannot be used with many-to-one/one-to-one relationships and/or uselist=False
```

without `lazy="write_only" this relationship returns collection and produces sql:
```
 SELECT document.*
	FROM document JOIN company AS company_1 ON document.org_id = company_1.id
	WHERE '123123' = company_1.inn
```

My aim to have lazy (or dynamic relationship) to use additional filter criteria on eager loading or when I want to filter on large collections for example:
```
company.documents.select().where(Document.is_active)
```

I've checked the place where sql raises error (https://github.com/sqlalchemy/sqlalchemy/blob/main/lib/sqlalchemy/orm/writeonly.py#L450) and found that `self.parent_property.direction` in this case has value `interfaces.MANYTOONE`

**Comment by zzzeek** [comment]
can you correct your example?  what is `Contractor` ?  `primaryjoin=foreign(Contractor.inn) == orgs.c.inn`

overall, if you expect Contractor -> Document to be a one-to-many collection, you would need the foreign key to be indicated on the Document side.

**Reply by artofhuman** [reply]
Sorry, I meant Company model.

**Reply by zzzeek** [reply, marked as answer]
so when you have A->B   , if column A.x is "foreign", that means this:


```
                      B.x = 1
A.x  =  1
A.x  =  1
A.x  =  1
A.x  =  1
A.x  =  1

                      B.x = 2
A.x  =  2
A.x  =  2
A.x  =  2
A.x  =  2
A.x  =  2

```

that is, for every `B`, there are zero or more `A` which contain a reference to `B`.   so A->B is many to one.   that's why your mapping shows this

**Reply by artofhuman** [reply]
Thanks, foreign annotation was in the wrong place.
